pkg/prometheus/promcfg.go

```diff
@@ -1868,6 +1876,28 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 		})
 	}
 
+	// Specific configuration generated for Prometheus Agent in DaemonSet mode.
+	if cg.PromAgentDaemonSet {
+		k8sSDConfig = cg.AppendMapItem(k8sSDConfig, "selectors", []yaml.MapSlice{
+			{
+				{
+					Key:   "role",
+					Value: "pod",
+				},
+				{
+					Key:   "field",
+					Value: "spec.nodeName=$(NODE_NAME)",
+				},
+			},
+		})
+		k8sSDConfig = cg.AppendMapItem(k8sSDConfig, "attach_metadata", yaml.MapSlice{
+			{
+				Key:   "node",
+				Value: true,
+			},
+		})
```

Comment:
I don't remember why we need to attach node metadata?

Reply:
Yeah we never discussed about this. I added this from reading Prometheus config doc. At first I thought we needed to set this to get node information. But now reread the doc, seems like this is only needed if we want to "attach" node info to targets, which we don't need right?

Reply:
yes if users want to retrieve node metadata, it's ok and can be done setting `attachMetadata.node: true` in the endpoint definition but it shouldn't be mandatory.